### PR TITLE
Correct interface label positioning during adding the line between nodes #409

### PR DESCRIPTION
**Pull request by piotrpekala7**
Ref. #409

### CODE REVIEW

File: src/app/services/link.service.ts [hunk 1]
```diff
@@ -20,12 +20,26 @@ export class LinkService {
         {
           node_id: source_node.node_id,
           port_number: source_port.port_number,
-          adapter_number: source_port.adapter_number
+          adapter_number: source_port.adapter_number,
+          label: {
```
Comment (piotrpekala7) [inline src/app/services/link.service.ts]: Due to current server implementation we're forced to pass all label data

**Comment by piotrpekala7** [comment]
We have to choose between 2 wrong solutions (IMO):
1) generating label data by UI side
2) add link then call updating labels

File: src/app/components/drawings-listeners/link-created/link-created.component.ts [hunk 2]
```diff
@@ -35,12 +35,45 @@ export class LinkCreatedComponent implements OnInit, OnDestroy {
...
+
+    let xLabelSourceNode = 0;
+    let yLabelSourceNode = 0;
+    let xLabelTargetNode = 0;
+    let yLabelTargetNode = 0;
+
+    if ((linkCreated.sourceNode.x <= linkCreated.targetNode.x) && (linkCreated.sourceNode.y <= linkCreated.targetNode.y)) {
+      xLabelSourceNode = Math.floor(linkCreated.sourceNode.width/2) + Math.round(x) + 5;
+      yLabelSourceNode = Math.floor(linkCreated.sourceNode.height/2) + Math.round(y) + 5;
+      xLabelTargetNode = Math.floor(linkCreated.targetNode.width/2) - Math.round(x) - 15;
```
Comment (piotrpekala7) [inline src/app/components/drawings-listeners/link-created/link-created.component.ts]: These values (5, 15) should moved to constants with other, I prefer to do that in another pr

**Comment by piotrpekala7** [comment]
Capture
this is how it works now

**Comment by ziajka** [comment]
Firefox & Linux, new line added:

Screenshot_2019-06-11_11-15-56

File: src/app/components/drawings-listeners/link-created/link-created.component.ts (hunk 2 above)
Comment (ziajka) [reply src/app/components/drawings-listeners/link-created/link-created.component.ts]: @piotrpekala7 makes sens, thanks

**Comment by ziajka** [comment]
+ & - @piotrpekala7

**Comment by piotrpekala7** [comment]
links

**Comment by ziajka** [comment]
@piotrpekala7  beautiful :) Thank you!
